### PR TITLE
build(prettier): Upgrade prettier to 1.7.4

### PR DESCRIPTION
**Pull request by billyvg**

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Changes to build requirements</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: docs-ui/components/confirm.stories.js [hunk 1]
```diff
@@ -9,16 +9,15 @@ import Button from 'sentry-ui/buttons/button';
 storiesOf('Confirm/Confirm', module).add(
   'default',
   withInfo({
-    text: 'Component whose child is rendered as the "action" component that when clicked opens the "Confirm Modal"',
-    propTablesExclude: [Button]
+    text:
+      'Component whose child is rendered as the "action" component that when clicked opens the "Confirm Modal"',
```
Comment (EvanPurkhiser) [inline docs-ui/components/confirm.stories.js]: 😢

Comment (billyvg) [reply docs-ui/components/confirm.stories.js]: yeah damn, using template strings forces it to be inline

Comment (billyvg) [reply docs-ui/components/confirm.stories.js]: Related:
https://github.com/prettier/prettier/issues/2889  
https://github.com/prettier/prettier/issues/2482

File: src/sentry/lint/engine.py [hunk 2]
```diff
@@ -113,7 +123,7 @@ def js_lint(file_list=None):
     return has_errors
 
 
-PRETTIER_VERSION = "1.2.2"
+PRETTIER_VERSION = "1.5.3"
```
Comment (MaxBittker) [inline src/sentry/lint/engine.py]: 1.7.4?

**Comment by MaxBittker** [comment]
small nit, it would be nice if the config changes and the format changes were different commits here (not saying you have to go back and fix)

File: src/sentry/lint/engine.py (hunk 2 above)
Comment (billyvg) [reply src/sentry/lint/engine.py]: Hmm doesn't look like we use this at all

File: src/sentry/lint/engine.py [hunk 3]
```diff
@@ -213,6 +220,46 @@ def js_test(file_list=None):
     return has_errors
 
 
+def less_format(file_list=None):
+    """
+    We only format less code as part of this pre-commit hook. It is not part
+    of the lint engine.
+    """
+    project_root = os.path.join(os.path.dirname(__file__), os.pardir, os.pardir, os.pardir)
+    prettier_path = os.path.join(project_root, 'node_modules', '.bin', 'prettier')
```
Comment (MaxBittker) [inline src/sentry/lint/engine.py]: maybe some of these variables should be lifted up for DRYness but just a nit

**Comment by billyvg** [comment]
@MaxBittker they are 
image

File: src/sentry/lint/engine.py (hunk 2 above)
Comment (MaxBittker) [reply src/sentry/lint/engine.py]: if this was working for you it means our " Prettier is out of date:" message isn't working, right?

**Review by MaxBittker — CHANGES_REQUESTED** [review]
IMO this is a nice improvement and we should merge it soon! the prettier_version thing seems like something may be broken though? would be bad if people were stuck running conflicting versions

**Comment by MaxBittker** [comment]
i also think we should "play test" checking this out on an existing dev environment and make sure that people won't be stuck in a weird state if they don't sentry upgrade right away (basically making sure that it fails)

File: src/sentry/lint/engine.py (hunk 3 above)
Comment (billyvg) [reply src/sentry/lint/engine.py]: afaik it doesn't look like `PRETTIER_VERSION` was ever used, the out of date message checks the version you have installed in `node_modules` vs the version defined in `package.json`

**Review by MaxBittker — APPROVED** [review]
💅 awesome! Lets make sure to help people with merge conflicts and upgrading so they don't get too annoyed 😅

**Pull request merged**